apps/oxlint/src-js/plugins/tokens.ts

```diff
@@ -181,10 +181,34 @@ export function initTokens() {
 function initTokensWithComments() {
   debugAssertIsNonNull(tokens);
   debugAssertIsNonNull(comments);
-  // TODO: `tokens` and `comments` are already sorted, so there's a more efficient algorithm to merge them.
-  // That'd certainly be faster in Rust, but maybe here it's faster to leave it to JS engine to sort them?
   // TODO: Replace `range[0]` with `start` once we have our own tokens which have `start` property.
-  tokensWithComments = [...tokens, ...comments].sort((a, b) => a.range[0] - b.range[0]);
+  tokensWithComments = [];
+
+  let tokensIndex = 0,
```

Comment:
Add before loop:

```ts
const tokensLength = tokens.length
  commentsLength = comments.length;
```

and then use those vars in the rest of the function, to avoid repeated property lookups.